components/SearchBar.tsx

```diff
@@ -0,0 +1,150 @@
+"use client";
+import { usePathname, useRouter } from "next/navigation";
+import Search from "@/components/filters/Search";
+import Sort from "@/components/filters/Sort";
+import format from "date-fns/format";
+import DateRangePicker from "@/components/DateRangePicker";
+import RoleFilter from "@/components/filters/RoleFilter";
+import { parseDateRangeSearchParam } from "@/lib/utils";
+import { SORT_BY_OPTIONS, FILTER_BY_ROLE_OPTIONS } from "@/lib/const";
+import { PageProps } from "@/lib/types";
+import { env } from "@/env.mjs";
+import { useDebouncedCallback } from "use-debounce";
+
+const SortOptions = Object.entries(SORT_BY_OPTIONS).map(([value, text]) => ({
+  value,
+  text,
+}));
+
+export const RoleOptions = Object.entries(FILTER_BY_ROLE_OPTIONS).map(
+  ([value, text]) => ({
+    value,
+    text,
+  }),
+);
+
+type SearchOptions = {
+  isSearch?: boolean;
+  isDaterange?: boolean;
+  isRoleFilter?: boolean;
+  isSort?: boolean;
+};
```

Comment:
I meant to make the "Search text box" alone to be reusable component, not the entire leaderboard filters.

Reply:
Okay , I'll correct it

Reply:
yup!